### ISSUE TITLE
Loss of precision curve values

### ISSUE THREAD

**Issue by markblumenau** [ultralytics/ultralytics#16883]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Val

### Bug

I was trying to validate my model, and for some reason validation with plots=False results in a loss of precision values in the precision-recall curve. plots=True has them, but i see no reason why it should change the behavior and it also doesn't affect other curves. Also plots=False results in some exception in threading.

### Environment

Ultralytics 8.3.9 🚀 Python-3.12.5 torch-2.4.1+cu124 CUDA:0 (NVIDIA GeForce RTX 3060, 12029MiB)
Setup complete ✅ (32 CPUs, 62.7 GB RAM, 88.1/490.2 GB disk)

OS                  Linux-6.8.0-45-generic-x86_64-with-glibc2.39
Environment         Linux
Python              3.12.5
Install             pip
RAM                 62.70 GB
Disk                88.1/490.2 GB
CPU                 AMD Ryzen 9 5950X 16-Core Processor
CPU count           32
GPU                 NVIDIA GeForce RTX 3060, 12029MiB
GPU count           1
CUDA                12.4

numpy               ✅ 2.1.2>=1.23.0
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.4.1+cu124>=1.8.0
torchvision         ✅ 0.19.1+cu124>=0.9.0
tqdm                ✅ 4.66.5>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.9>=2.0.0
torch               ✅ 2.4.1+cu124!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

```python
from ultralytics import YOLO
yolo = YOLO("path_to_my_model")
results = yolo.val(data="path_to_my_data_yaml", plots=False)
print(len(results.curves_results[0][0]) , len(results.curves_results[0][1]))
```

### Additional

Without plots=False
![image](https://github.com/user-attachments/assets/cc21598e-7fa8-4604-a6f5-defd40e03045)

With
![image](https://github.com/user-attachments/assets/e77f707a-e46a-441d-ab93-ba19825f9bbe)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @markblumenau, thank you for reaching out and sharing your issue with Ultralytics 🚀! This is an automated response to help get you started. An Ultralytics engineer will also look into your issue soon.

We recommend checking out the [Documentation](https://docs.ultralytics.com) for more insights, especially if you're new to our repository. You may find useful examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage there that might address your concerns.

Since you've reported this as a 🐛 Bug, please ensure you provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to facilitate the debugging process.

Also, if this issue involves custom training ❓ questions, kindly provide detailed information including dataset examples and training logs. Make sure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our Ultralytics community to engage further:
- Real-time chat on [Discord](https://ultralytics.com/discord) 🎧
- Explore discussions on [Discourse](https://community.ultralytics.com)
- Share and learn on our [Subreddit](https://reddit.com/r/ultralytics)

## Upgrade

Please ensure that you have upgraded to the latest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can use YOLO across the following environments that come with pre-installed dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: 
<a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> 
<a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> 
<a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Refer to the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) 
<a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

A green badge indicates that all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, verifying the correct functioning of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

**Comment by Y-T-G** [comment]
Can you reproduce this with the `ultralytics` supported datasets?

https://docs.ultralytics.com/datasets/
